debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/buffered/rocksdb/RocksDbLogMinerTransactionCache.java

```diff
@@ -0,0 +1,596 @@
+
+    // Use single column family for all transaction events (performance improvement)
+    private ColumnFamilyHandle eventsColumnFamily;
+    private ColumnFamilyHandle rowIdIndexColumnFamily; // Secondary index for rowId lookups
+    private ColumnFamilyHandle transactionsColumnFamily; // Stores transaction metadata (username etc.)
+
+    // Heap-backed cache for quick access to specific metadata
+    private final Map<String, TreeSet<Integer>> eventIdsByTransactionId = new HashMap<>();
+
+    // Batch flushing for better performance
+    private final int batchFlushThreshold;
+    private int pendingWrites = 0;
+
+    @SuppressWarnings("resource")
+    public RocksDbLogMinerTransactionCache(String dbPath, Options options, DBOptions dbOptions, int batchFlushThreshold) {
+        this.dbOptions = dbOptions;
+        this.batchFlushThreshold = batchFlushThreshold;
+        // Disable WAL since this is a spillover cache - data can be regenerated from source
+        this.writeOptions = new WriteOptions().setDisableWAL(true);
+        initializeDatabase(dbPath, options, dbOptions);
+        // Heap cache is ephemeral and built on-demand as transactions are added
+    }
+
+    private void initializeDatabase(String dbPath, Options options, DBOptions dbOptions) {
+        try {
+            List<ColumnFamilyDescriptor> existingCFDescriptors = new ArrayList<>();
+            List<ColumnFamilyHandle> existingCFHandles = new ArrayList<>();
+
+            try {
+                List<byte[]> existingCFNames = RocksDB.listColumnFamilies(options, dbPath);
+                if (existingCFNames.isEmpty()) {
+                    existingCFDescriptors.add(new ColumnFamilyDescriptor(RocksDB.DEFAULT_COLUMN_FAMILY, new ColumnFamilyOptions()));
+                }
+                else {
+                    for (byte[] cfName : existingCFNames) {
+                        existingCFDescriptors.add(new ColumnFamilyDescriptor(cfName, new ColumnFamilyOptions()));
+                    }
+                }
+            }
+            catch (RocksDBException e) {
+                existingCFDescriptors.add(new ColumnFamilyDescriptor(RocksDB.DEFAULT_COLUMN_FAMILY, new ColumnFamilyOptions()));
+            }
+
+            this.db = RocksDB.open(dbOptions, dbPath, existingCFDescriptors, existingCFHandles);
+
+            ColumnFamilyHandle eventsHandle = null;
+            ColumnFamilyHandle indexHandle = null;
+            ColumnFamilyHandle txHandle = null;
+
+            for (ColumnFamilyHandle handle : existingCFHandles) {
+                String name = new String(handle.getName(), StandardCharsets.UTF_8);
+                if ("events".equals(name)) {
+                    eventsHandle = handle;
+                }
+                else if ("rowIdIndex".equals(name)) {
+                    indexHandle = handle;
+                }
+                else if ("transactions".equals(name)) {
+                    txHandle = handle;
+                }
```

Comment:
Could we construct 3 static member variables that represent the `byte[]` for each of the strings `events`, `rowIdIndex`, and `transactions` allowing us to just compare the `byte[]` during the loop without allocating an extra `String`?